examples/sycl/pvc/dual_gemm/tensor_silu.h

```diff
@@ -0,0 +1,148 @@
+ *
+ **************************************************************************************************/
+#pragma once
+
+// Cutlass includes
+#include "cutlass/cutlass.h"
+#include "cutlass/tensor_view.h"
+
+#include "cutlass/util/reference/device/tensor_foreach.h"
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace cutlass {
+namespace reference {
+namespace device {
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace detail {
+
+template <
+  typename Element,               ///< Element type
+  typename Layout>                ///< Layout function
+struct TensorSiLuFunc {
+
+  /// View type
+  using TensorView = TensorView<Element, Layout>;
+
+  /// Coordinate in tensor's index space
+  using TensorCoord = typename TensorView::TensorCoord;
+
+  /// Parameters structure
+  struct Params {
+
+    //
+    // Data members
+    //
+
+    TensorView view_in0, view_in1, view_out;
+    //
+    // Methods
+    //
+
+    Params(
+      TensorView view_out_ = TensorView(),
+      TensorView view_in0_ = TensorView(),
+      TensorView view_in1_ = TensorView()
+    ):
+      view_out(view_out_), view_in0(view_in0_), view_in1(view_in1_){
+
+    }
+  };
+
+  //
+  // Data members
+  //
+
+  Params params;
+
+  //
+  // Methods
+  //
+
+  CUTLASS_DEVICE
+  TensorSiLuFunc(Params const &params): params(params) {
+
+  }
+
+  CUTLASS_DEVICE
+  void operator()(TensorCoord const &coord) {
+
+    Element const& in0_val = params.view_in0.at(coord);
+    Element const& in1_val = params.view_in1.at(coord);
+
+    cutlass::epilogue::thread::SiLu<Element> silu;
+    cutlass::multiplies<Element> mul;
+
+    auto silu_lhs = silu(in0_val);
+    params.view_out.at(coord) = mul(silu_lhs, in1_val);
+  }
+};
+
+} // namespace detail
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// Apply SiLu on a tensor
+template <
+  typename Element,               ///< Element type
+  typename Layout>                ///< Layout function
+void TensorSiLu(
+  TensorView<Element, Layout> view_out,       ///< destination tensor
+  TensorView<Element, Layout> view_in0,       ///< source tensor
+  TensorView<Element, Layout> view_in1) {        ///< source tensor
+  
+  using Func = detail::TensorSiLuFunc<Element, Layout>;
+  using Params = typename Func::Params;
+
+  TensorForEach<Func, Layout::kRank, Params>(
+    view_out.extent(),
+    Params(view_out, view_in0, view_in1)
+  );
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+} // namespace device
+} // namespace reference
+} // namespace cutlass
+
+#if (CUTLASS_ENABLE_SYCL)
+namespace sycl {
```

Comment:
I think this is needed because TensorView doesn't have a default copy constructor, could you try the change I've made here https://github.com/FMarno/cutlass-fork/blob/5a69f1944a5ce5b460aefd25a1c63acea6e8115b/include/cutlass/tensor_view.h#L149

Reply:
Yeah this worked. Thank you.